lib/core/src/sdk.rs

```diff
@@ -162,17 +174,22 @@ impl LiquidSdkBuilder {
         self
     }
 
-    pub fn build(&self) -> Result<Arc<LiquidSdk>> {
+    fn get_working_dir(&self) -> Result<String> {
+        let fingerprint_hex: String =
+            Xpub::decode(self.signer.xpub()?.as_slice())?.identifier()[0..4].to_hex();
+        self.config
+            .get_wallet_dir(&self.config.working_dir, &fingerprint_hex)
+    }
+
+    pub(crate) fn build(&self) -> Result<Arc<LiquidSdk>> {
         if let Some(breez_api_key) = &self.config.breez_api_key {
             LiquidSdk::validate_breez_api_key(breez_api_key)?
         }
 
-        fs::create_dir_all(&self.config.working_dir)?;
+        #[cfg(not(all(target_family = "wasm", target_os = "unknown")))]
+        std::fs::create_dir_all(&self.config.working_dir)?;
```

Comment:
I think it would also be nice to move this code to an upper layer. Perhaps in the connect_with_signer ?

Reply:
Done in [7e85942](https://github.com/breez/breez-sdk-liquid/pull/792/commits/7e859427927bdbb2c9c6ee94531af9c855ff3e31). I also moved the log print header as it was not being called when integrators provided their own signer.